teps/0123-specifying-on-demand-retry-in-pipelinetask.md

```diff
@@ -0,0 +1,183 @@
+  `on-demand-retry` is waiting to be restarted. For example, the following pipeline has `B` defined as
+  `on-demand-retry` and B fails while `D` is running, the `pipeline` continues to schedule `E` and `F` while waiting
+  on user input for `B`. If there is a `finally` section in the pipeline, it will wait until `B` is retried and succeeds.
+  The user has an option to stop retrying and execute finally by [gracefully cancelling a pipelineRun](How can the user request to stop retrying and execute finally).
+
+```markdown
+   ┌─────────────────────────────────────┐ ┌────────────┐
+   │                          Main       │ │  Finally   │
+   │   ┌───┐    ┌───┐                    │ │            │
+   │   │   │    │   │                    │ │            │
+   │   │ A ├────┤ B (failed)             │ │            │
+   │   │   │    │ - │                    │ │            │
+   │   └───┘    └───┘                    │ │   ┌───┐    │
+   │                                     │ │   │   │    │
+   │                                     │ │   │ G │    │
+   │   ┌───┐    ┌───┐    ┌───┐    ┌───┐  │ │   │   │    │
+   │   │   │    │   │    │   │    │   │  │ │   └───┘    │
+   │   │ C ├────┤ D ├────┤ E ├────┤ F │  │ │            │
+   │   │   │    │   │    │   │    │   │  │ │            │
+   │   └───┘    └───┘    └───┘    └───┘  │ │            │
+   │                                     │ │            │
+   └─────────────────────────────────────┘ └────────────┘
+```
+
+### Use Cases
+
+#### CD Use Case
+
+Let's take an example of a CD `pipeline` to deploy an application to production. The CD `pipeline` is configured to
+trigger a new run for every new change request i.e. a PR created in the `application-deployment` repo.
+
+The release manager creates a new change request by creating a PR with the details needed for the deployment such as
+an application GitHub repo and a specific branch or commit. Creation of a PR triggers a new `pipelineRun` in which the
+application source is cloned, built, and deployed to the production cluster. After the deployment succeeds, a flaky
+acceptance test is executed against the production deployment. The acceptance tests are flaky and results in failure
+sometimes. This failure prevents updating the change request with the deployment evidences.
+
+Now, the release manager has three choices to work around this flakiness:
+
+1) Configure acceptance-tests task with a reasonable amount of retries. But this might or might not work depending on the
+cause of the flakiness. Retrying a task once it fails might address network connectivity issue as mentioned in the pipeline
+[specification](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#using-the-retries-field). If the
+flakiness is caused by an issue which requires a fix from the users, this kind of retry does not help.
+
+2) Close the existing PR and create a new PR for the same request. This new PR will trigger a new `pipelineRun` which
```

Comment:
I don't think close/re-open are the only alternatives. Pushing a new commit or a /retest like chat command can both work here.

Reply:
I agree that closing/re-opening is not the only option here. 

I think @pritidesai's point was not about the specific way the pipeline is triggered, but rather that a completely new `PipelineRun` is required. On-demand retries of a task would change things so that the `PipelineRun` would not terminate right away upon failure, giving the engineer an option to fix the underlying condition and retry the specific task without having to repeat the execution of the rest of the pipeline.